src/bindings/python/constraints.txt

```diff
@@ -17,7 +17,7 @@ patchelf<=0.17.2.1
 # Frontends
 docopt~=0.6.2
 paddlepaddle==2.5.2
-tensorflow>=1.15.5,<2.15.0
+tensorflow>=1.15.5,<2.16.0
```

Comment:
wondering why TF is a dependency of Python API?

Reply:
@popovaan, do you know if we added this for tf.Decoder?

Reply:
we can move framework dependent tests here https://github.com/openvinotoolkit/openvino/tree/master/tests/layer_tests/py_frontend_tests to keep Python API tests with minimum set of dependencies

It will allow to decouple Python API building and testing from frameworks and support newer Python versions ealier than it's done by frameworks. Otherwise, we have those dependencies which are hard to resolve.